### ISSUE TITLE
the --24-bit-color option is unusable from .gitconfig

### ISSUE THREAD

**Issue by lepotic** [dandavison/delta#567]
```
% git config --global delta.24-bit-color yes
error: invalid key: delta.24-bit-color
```

OTOH,
```
% git config --global delta.true-color yes
```

Manually editing `~/.gitconfig` didn't help either:

```gitconfig
[delta]
    24-bit-color = yes
```

```
% git log
fatal: bad config line 18 in file /home/redacted/.gitconfig
```

Seems like git doesn't accept keys that start with a digit.

**Comment by Kr1ss-XD** [comment]
FWIW I'm using a workaround for this issue. It's possible to set e.g.
```gitconfig
[core]
	pager	= delta --24-bit-color=always
```
instead. Hope this helps, _cheers_ !

**Comment by dandavison** [comment]
Hi @lepotic, @Kr1ss-XD, hm, yes given the importance of gitconfig to delta perhaps (as you suggest) we should create an alias for this option which was poorly named in retrospect (I think it came before gitconfig support FWIW).

**Comment by Kr1ss-XD** [comment]
> [...] we should create an alias [...]

@dandavison @lepotic what about `--true-color=[always|auto|never]` then ?

**Comment by dandavison** [comment]
> @dandavison @lepotic what about --true-color=[always|auto|never] then ?

Done. https://github.com/dandavison/delta/pull/571 `24-bit-color` is now marked as deprecated in `--help`. If `true-color` is at its default value, and you supply a differing value for `24-bit-color`, then that will take effect.

**Comment by dandavison** [comment]
By the way, are your terminal emulators not setting `COLORTERM`? Delta will automatically support 24-bit color without having to use `--true-color`, as long as our terminal emulators are advertizing their capability via `COLORTERM=truecolor` or `COLORTERM=24bit`. Seeing as other applications may respond to this, it might be a better solution to set that env var than to put delta into `always` mode.

**Comment by lepotic** [comment]
> By the way, are your terminal emulators not setting COLORTERM? @dandavison 

Mine didn't (st). I had to set it myself.

There seems to be a way of querying the terminal capabilities, described in https://gist.github.com/XVilka/8346728#querying-the-terminal (You probably know this gist, but anyway.)

**Comment by Kr1ss-XD** [comment]
Thx @dandavison !

In regards to `$COLORTERM`, I've noticed that some terminal emulators don't set this, despite being 24-bit capable. Examples are [xterm](https://invisible-island.net/xterm/) and [wezterm](https://wezfurlong.org/wezterm/), while [xfce4-terminal](https://docs.xfce.org/apps/terminal/start), [kitty](https://sw.kovidgoyal.net/kitty/) and [alacritty](https://github.com/alacritty/alacritty) correctly set the variable to `truecolor`.

Tmux, if I recall this correctly, needs to be configured to take advantage of true color. If it is, it does set `$COLORTERM` though, even when run inside xterm or wezterm.

Long story short, it might make sense to use delta's flag anyways, when using multiple terminal emulators.

**Comment by nick87720z** [comment]
This options seems not in gitconfig support.
```
delta --true-color always
```
 works, but
```
git config --global delta.true-color always
```
has no effect.
